### DISCUSSION TITLE
Binding key presses to custom methods

### DISCUSSION THREAD

**Discussion by jeffwright13** [Textualize/textual#310] [Q&A]
Is there a way to bind keypresses (or any other action, actually) to custom methods? All the examples I've seen are bound to single actions, defined in quotes within an `await` statement. What I'd like to do is explore stringing several actions in a row, as defined in a method. For example, let's say I have individual bindings for keypresses "F", "M", E" and "P", each of which toggles (hids/unhides) its own widget. I'd like to define  a series of actions to be executed on keypress "A" which would toggle all four of "F", "M", E" and "P".

**Comment by jeffwright13** [comment]
Actually I just stumbled on the "test app" code in `app.py` (`MyApp`). I can see there's a keybinding there defined as:
```
await self.bind("b", "toggle_sidebar", "Toggle sidebar")
```
...and there is also an `action_toggle_sidebar` method defined below it. So that seems like a promising breadcrumb? At the very least, the magic `action_` prefix seems to turn my defined binding into one that is callable, so I'l continue down this path for a bit and see where it leads.

**Reply by aitoehigie** [reply]
You seem to have correctly answered the question you asked.

Kudos
